src/core/thread/link_quality.cpp

```diff
@@ -206,7 +225,7 @@ uint8_t LinkQualityInfo::ConvertRssToLinkQuality(int8_t aNoiseFloor, int8_t aRss
 
 int8_t LinkQualityInfo::ConvertLinkQualityToRss(int8_t aNoiseFloor, uint8_t aLinkQuality)
 {
-    uint8_t linkmargin = 0;
+    int8_t linkmargin = 0;
```

Comment:
according to spec, LinkMargin's value might be in range `[0, 130]`, the type `int8_t` should not be enough to represent up to value 130.

Reply:
This was changed in #5505 to fix fuzzer error. My PR apply the change for CI before that PR merged.

In this function, `linkmargin` is at most `kLinkQuality3LinkMargin`(50). If `linkmargin + aNoiseFloor` is greater than `128`(which is impossible since RSS should be negative), as the return type is `int8_t`, we would get the same result even if use `uint8_t`.

So the true problem is(not caused by this PR), the range of RSS is `[-130, 0]`, but `int8_t` is not enough for it. However I think in this function, as `linkmargin` is alwasy positive, the result of `linkmargin + aNoiseFloor` is always within the valid range of `int8_t`.